### ISSUE TITLE
ImportError: No module named twisted.internet.protocol

### ISSUE THREAD

**Issue by Makishima** [Laharah/deluge-uTorrentImport#2]
Hello!
Trying to use script this way 
> C:\Python26>python "C:\Program Files (x86)\Deluge\deluge-1.3.11-py2.6.egg\uTorrent_to_Deluge.py" --host 127.0.0.1 --port 58846 "C:\Users\Nikolay\AppData\Roaming\uTorrent\resume.dat"

Ang get this error
> Traceback (most recent call last):
  File "C:\Program Files (x86)\Deluge\deluge-1.3.11-py2.6.egg\uTorrent_to_Deluge.py", line 11, in <module>
    from deluge.ui.client import client
  File "deluge\ui\client.py", line 36, in <module>
ImportError: No module named twisted.internet.protocol

Python version 
> 2.6.6 (r266:84297, Aug 24 2010, 18:13:38) [MSC v.1500 64 bit (AMD64)]

**Comment by Laharah** [comment]
you have to have twisted installed as a package in your python installation.

You can either try `pip install twisted`, or you can download and install it manually here: https://twistedmatrix.com/trac/wiki/Downloads

**Comment by Makishima** [comment]
Im sorry for late response
But after installing twisted there new error
> C:\Python26>python "C:\Program Files (x86)\Deluge\deluge-1.3.11-py2.6.egg\uTorrent_to_Deluge.py" --host 127.0.0.1 --port 58846 "C:\Users\Nikolay\AppData\Roaming\uTorrent\resume.dat"
Traceback (most recent call last):
  File "C:\Program Files (x86)\Deluge\deluge-1.3.11-py2.6.egg\uTorrent_to_Deluge.py", line 11, in <module>
    from deluge.ui.client import client
  File "deluge\ui\client.py", line 36, in <module>
  File "C:\Python26\lib\site-packages\twisted\internet\protocol.py", line 15, in <module>
    from zope.interface import implements
ImportError: No module named zope.interface

**Comment by Laharah** [comment]
that's a common issue. Twisted requires zope to be installed as well but some twisted instalations don't come with it. 

The easiest way is to install pip (assuming you don't have it installed already). Follow the instructions here: [https://pip.pypa.io/en/latest/installing.html#install-pip](https://pip.pypa.io/en/latest/installing.html#install-pip)

basically, just download the get-pip.py, run it with `python get-pip.py`.

once that's done type `pip install zope`

**Comment by Makishima** [comment]
So I installed pip via ez_setup for WIndows, log here http://pastebin.com/aJytBtqn
But still get that error 
> PS C:\Python26> .\python.exe "C:\Program Files (x86)\Deluge\deluge-1.3.11-py2.6.egg\uTorrent_to_Deluge.py" --host 127.0.
0.1 --port 58846 "C:\Users\Nikolay\AppData\Roaming\uTorrent\resume.dat"
Traceback (most recent call last):
  File "C:\Program Files (x86)\Deluge\deluge-1.3.11-py2.6.egg\uTorrent_to_Deluge.py", line 11, in <module>
    from deluge.ui.client import client
  File "deluge\ui\client.py", line 36, in <module>
  File "C:\Python26\lib\site-packages\twisted\internet\protocol.py", line 15, in <module>
    from zope.interface import implements
ImportError: No module named zope.interface
PS C:\Python26> .\python.exe "C:\Program Files (x86)\Deluge\deluge-1.3.11-py2.6.egg\uTorrent_to_Deluge.py" --host 127.0.0.1 --port 58846 "C:\Users\Nikolay\AppData\Roaming\uTorrent\resume.dat"
Shame on me but Im not very familiar with python.

**Comment by Laharah** [comment]
No problem.

Did you install zope after you installed pip? pip is used for easily 
installing python packages.

Make sure you run `pip install zope`

**Comment by Makishima** [comment]
There a few more dependents
> File "C:\Python26\lib\site-packages\twisted\internet\ssl.py", line 23, in <module>
   from OpenSSL import SSL

and

>  File "deluge\common.py", line 44, in <module>
ImportError: No module named chardet

Even after installing them script says that
> C:\Python26\lib\site-packages\cryptography\__init__.py:25: DeprecationWarning: Python 2.6 is no longer supported by the Python core team, please upgrade your Python.
  DeprecationWarning
Connection failed!
Please ensure that deluge is running and is in client/server mode, not "classic" mode.

But it running in that mode

**Comment by Laharah** [comment]
hmmm, well the script is at least running now, but it seems that it can't connect to the deluged daemon. I'm sorry this is turning into such a hassle for you.

Double check that the "classic" checkbox in settings>Interface is unchecked, And then restart deluge. Then just to be sure, check that "deluged" is running in the windows task manager (go to "details" after hitting `ctrl-shift-esc`) 

After that make sure you're using the port from settings>daemon port. While you're there make sure "allow remote connections" is checked.

If everything is set up right and it's still not working, re-download the script. I pushed an update that will log the connection result on failure. It might help me figure out what's going wrong.

I know the cryptography package was recently updated but I don't think that the depreciation warning is causing the failure, If it is I'll have to update the script into a plugin or something.

**Comment by Makishima** [comment]
something wrong with last changes because script just write
> C:\Python26\lib\site-packages\cryptography\__init__.py:25: DeprecationWarning: Python 2.6 is no longer supported by the Python core team, please upgrade your Python.
  DeprecationWarning

and do nothing even shutdown itself

**Comment by Laharah** [comment]
well, damn, I'm completely out of ideas, I changed one line and it showed the connection error without trouble when I tested it. 

Tell you what. give me a day or so and I'll turn the script into a super basic plugin prototype. It should be enough to at least get your torrent's transferred. Once it works in the most basic form, I'll round it out to a 2.0 release.

**Comment by Makishima** [comment]
Thank you.
I will be happy to offer any help if needed.

**Comment by Laharah** [comment]
Well, my Friday night plans fell through, so I coded up the basic version of plugin. It's pretty stripped down, but it has all the essentials, plus it auto-searches for the resume.dat file so you don't have to fill it in yourself. It doesn't report anything to the user though, only to the debug logs so if anything gets skipped you'd have to read the debug logs to see it. 

I gave it a new name and I tested it on the newest utorrent for windows in both the 2.6 and 2.7 versions of deluge and they both worked without issue. I wasn't very rigorous with my tests so it's possible some torrents might give it problems, but I think I handled most of the edge cases correctly.

anyway you can get the alpha builds on my develop branch here: https://github.com/Laharah/deluge-uTorrentImport/tree/develop/dist

Just download the version you need, open deluge, install the plugin, turn it on, and hit the button in the bottom left; the torrents should start pouring in. They'll be paused and they should all be re-checking as well to be safe.

Let me know if you run into any issues, and I'll see if I can replicate the problem. I'll probably spend an hour or two this weekend rounding it out with reports and whatnot, so any bugs you can find before I finish it would be awesome.

**Comment by Makishima** [comment]
And again nothing :smile: Where logs is located?
So I have WIn8.1Pro; Deluge Client: 1.3.11, Server: 1.3.11, libtorrent: 0.16.18.0; uTorrent 3.4.3 build 40760 32-bit

**Comment by Laharah** [comment]
... of course. to get the debug logs, first stop the deluge daemon and quit deluge. Then open 2 command prompts to your deluge path (eg: `C:\program files\deluge`) then in one run `deluged-debug -L debug > deluged_log.txt` and in the other run `deluge-debug -L debug > deluge_log.txt` That will run both the daemon and the client in debug mode and capture the output into `deluged_log.txt` and `deluge_log.txt`

Boot those up, try running the plugin and then post them here or somewhere I can look over them and I'll see if I can figure out what it is about your setup my scripts hate so much.

**Comment by Laharah** [comment]
Wait, before you go through the debug logs (if you haven't already) I think I might have fixed the bug in the new plugin. I was a regression I forgot to port over to the plugin from the script involving relative paths in resume.dat that doesn't affect my setup. 

I added the line back in and I think the plugin should work now. I posted the new binaries on the dev branch again: https://github.com/Laharah/deluge-uTorrentImport/tree/develop/dist

See if it works with the new version (you might have to restart deluge after installing the updated version). If it's still broken send on the debug logs.
